python/pyspark/sql/pandas/serializers.py

```diff
@@ -371,3 +375,292 @@ def load_stream(self, stream):
                 raise ValueError(
                     "Invalid number of pandas.DataFrames in group {0}".format(dataframes_in_group)
                 )
+
+
+class ApplyInPandasWithStateSerializer(ArrowStreamPandasUDFSerializer):
+    def __init__(
+        self,
+        timezone,
+        safecheck,
+        assign_cols_by_name,
+        state_object_schema,
+        soft_limit_bytes_per_batch,
+        min_data_count_for_sample,
+        soft_timeout_millis_purge_batch,
+    ):
+        super(ApplyInPandasWithStateSerializer, self).__init__(
+            timezone, safecheck, assign_cols_by_name
+        )
+        self.pickleSer = CPickleSerializer()
+        self.utf8_deserializer = UTF8Deserializer()
+        self.state_object_schema = state_object_schema
+
+        self.result_state_df_type = StructType(
+            [
+                StructField("properties", StringType()),
+                StructField("keyRowAsUnsafe", BinaryType()),
+                StructField("object", BinaryType()),
+                StructField("oldTimeoutTimestamp", LongType()),
+            ]
+        )
+
+        self.result_state_pdf_arrow_type = to_arrow_type(self.result_state_df_type)
+        self.soft_limit_bytes_per_batch = soft_limit_bytes_per_batch
+        self.min_data_count_for_sample = min_data_count_for_sample
+        self.soft_timeout_millis_purge_batch = soft_timeout_millis_purge_batch
+
+    def load_stream(self, stream):
```

Comment:
method level comments here and everywhere else. specifically the parameters being untyped, a comment about the parameter type will be helpful.